crates/ruff_linter/src/rules/pylint/rules/property_with_parameters.rs

```diff
@@ -58,14 +63,7 @@ pub(crate) fn property_with_parameters(
     {
         return;
     }
-    if parameters
-        .posonlyargs
-        .iter()
-        .chain(&parameters.args)
-        .chain(&parameters.kwonlyargs)
-        .count()
-        > 1
```

Comment:
The advantage of defining your own iterator type (vs using chaining) is that you could implement `ExactSizeIterator` and you could then call `.len()` here.

But I think the better solution here is to add a method to `parameters` to get this length

Reply:
> But I think the better solution here is to add a method to `parameters` to get this length

Currently the rule is doing something a little odd: it's checking how many nonvariadic parameters there are in this function, but paying no attention to how many variadic parameters there are. That makes no sense to me based on what the rule is supposed to be checking: I think it makes just as much sense to emit an error if an `@property`-decorated function has `*args` or `**kwargs` parameters, as that's just as invalid.

I think maybe I'll just revert all changes to this file for now, and fix the bug (at least, I think it's a bug) in a separate PR, with added tests.

Reply:
Followup PR at #11200